### ISSUE TITLE
Move using_proxy to Credentials class

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#1490]
## Feature Description

Currently the `using_proxy` method is part of the `OAuth_Client` class.

The `OAuth_Client` class is only exposed through the `Authentication` class.

Since logic for `using_proxy` is entirely dependent on `Credentials` it makes sense to move this method to `Credentials`.

Current references to this method and what it would become:
```diff
- Authentication->OAuth_Client->using_proxy()
+ Authentication->Credentials->using_proxy()
```

```diff
- OAuth_Client->using_proxy()
+ OAuth_Client->Credentials->using_proxy()
```

```diff
- Debug_Data->Authentication->OAuth_Client->using_proxy()
+ Debug_Data->Credentials->using_proxy()
```

This helps flatten the chain to access this common data and no longer requires the `OAuth_Client` to be initialized to get it.

It also has the added benefit of making this data more readily available to other classes that already have access to `Credentials`.

E.g. in #1479 the Notifications class needs access to `using_proxy` - this becomes much more direct via `Credentials` which it already has an instance of.
```diff
- Notifications->Authentication->OAuth_Client->using_proxy()
+ Notifications->Credentials->using_proxy()
```

> Originally it seemed that `Authentication` would be the better place for this method but after closer evaluation, it seems to be a better fit on `Credentials`. This is a horizontal shift for classes that already access this method through `Authentication`.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `OAuth_Client::using_proxy` method is deprecated and calls `_deprecated_function` when called
	- Returns `OAuth_Client->Credentials->using_proxy()`
* The current logic for `using_proxy` is added to `Credentials` including tests
* All current usage is updated to call `using_proxy` via `Credentials`

## Implementation Brief

* Add `Credentials::using_proxy` method, basically:
```php
$creds = $this->get();
return (bool) preg_match( '/\.apps\.sitekit\.withgoogle\.com$', $creds['oauth2_client_id'] );
```
* Deprecate `OAuth_Client::using_proxy`
* Update all usage to call `using_proxy` via `Credentials`

## QA Brief

* 

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
IB ✅

**Comment by aaemnnosttv** [comment]
## QA ✅ 

* ⚠️ One instance of the deprecated use is present, although I believe this is due to #1211 being merged in after this one 😄   
https://github.com/google/site-kit-wp/blob/82d6fa72656d48de04d9c9b91462471ed28c0227/includes/Modules/Analytics.php#L695-L697  
I've fixed this in #1582 via https://github.com/google/site-kit-wp/pull/1582/commits/4242a5e1bc31941610c119306f2b1b9c1f574585 which also targets this release, so I think it's a non-issue here.
